packages/@aws-cdk/aws-sns/lib/topic-ref.ts

```diff
@@ -168,21 +173,27 @@ export abstract class TopicRef extends cdk.Construct implements events.IEventRul
   /**
    * Adds a statement to the IAM resource policy associated with this topic.
    *
-   * If this topic was created in this stack (`new Topic`), a topic policy
-   * will be automatically created upon the first call to `addToPolicy`. If
-   * the topic is improted (`Topic.import`), then this is a no-op.
+   * If this topic was created in this stack (``new Topic`), a topic policy
+   * will be automatically created upon the first call to ``addToPolicy``. If
+   * the topic is improted (``Topic.import``), then this is a no-op.
+   *
+   * @param statement the statement to add to the policy. It must have an Sid
+   *                  that has not yet been used in this resource policy.
    */
   public addToResourcePolicy(statement: iam.PolicyStatement) {
```

Comment:
Why do we require that the user supplied an SID. If there is no SID, why not allocate one for the user? What's the value in this?

Reply:
Immutability of the SID - I can guarantee that an SID is not used twice in a given Topic policy (that'd be illegal), and I can be suer it doesn't change after me. Of course, I could also do the validation at synthesis time in a `validate` function, but fail fast, right?